From a0075f2a27929c47f414fd4338bd299758919def Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Fri, 8 Aug 2025 11:16:18 +0200
Subject: a11y Windows: Implement UIA_AriaRolePropertyId for
 Accessible::Heading

This way also the property UIA_LocalizedControlTypePropertyId is
populated and screen readers like narrator can announce the role of
headings correctly.

Pick-to: 6.8 6.9 6.10
Change-Id: I745e4420bbe0f2cc1ef0e52ecdeab781d641b474
Reviewed-by: Michael Weghorn <m.weghorn@posteo.de>
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 6dfef0040307f9d78eaa4e1707bd6b58e22d961d)
---
 .../windows/uiautomation/qwindowsuiamainprovider.cpp          | 4 ++++
 1 file changed, 4 insertions(+)

diff --git x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
index 50f5f7b2c7f7d87d11152c6788ab118539f17b70..30471ee68c1eb7868edd21154fd8e35126e8f199 100644
--- x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
+++ y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
@@ -503,6 +503,10 @@ HRESULT QWindowsUiaMainProvider::GetPropertyValue(PROPERTYID idProp, VARIANT *pR
         // Accelerator key.
         *pRetVal = QComVariant{ accessible->text(QAccessible::Accelerator) }.release();
         break;
+    case UIA_AriaRolePropertyId:
+        if (accessible->role() == QAccessible::Heading)
+            *pRetVal = QComVariant{ QStringLiteral("heading") }.release();
+        break;
     case UIA_AriaPropertiesPropertyId:
         setAriaProperties(accessible, pRetVal);
         break;
